uaclient/security.py

```diff
@@ -562,6 +563,40 @@ def fix_security_issue_id(
     }
 
     if "CVE" in issue_id:
+        # Check livepatch status for CVE in fixes before checking CVE api
+        try:
+            status_stdout, _ = system.subp(
+                [
+                    "canonical-livepatch",
+                    "status",
+                    "--verbose",
+                    "--format=json",
+                ]
+            )
+        except exceptions.ProcessExecutionError:
+            pass
+        if status_stdout:
```

Comment:
Maybe I am missing something, but if we have an exception when calling the `canonical-livepatch` command, wouldn't the `status_stdout` variable be undefined here ?

We can confirm that by adding a test where raise an exception for that command and see how it proceeds. To make that test easier, we could create a function for that new livepatch check instead and then just test the function directly for the exception scenarios and not the whole `fix_security_issue_id`

Reply:
Hey, @lucasmoura thanks for the feedback! The direction I'd received from @orndorffgrant was that just proceeding with the usual flow for `fix_security_issue_id` if an exception is raised was fine. That's why I had the `except exceptions.ProcessExecutionError: pass` and `if status_stdout:` checks.

If the group thinks a separate function with separate tests is the better approach and within the scope of the issue, happy to pursue that line of thought!

Reply:
Ah sorry I missed this @dixonmyles. I think we can skip the refactor to move this into its own function for now, but @lucasmoura is right about the `status_stdout` variable. When the `canonical-livepatch` command fails, then we get this error:
```
UnboundLocalError: local variable 'status_stdout' referenced before assignment
```
The simplest fix to this is to define the variable before the `try` block. I suggest adding 
```python
status_stdout = None
```
right before the `try` on line 567

Reply:
@orndorffgrant @lucasmoura thank you both! I think given the back and forth, and the comment from @renanrodrigo, I'll look into adding a test case for this path as well